packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -276,168 +279,166 @@ export const initProvider = () => (dispatch: Dispatch) => {
     return decision.promise;
 };
 
-export const fetchMetadata =
-    (deviceStateArg?: string) => async (dispatch: Dispatch, getState: GetState) => {
-        const provider = selectSelectedProviderForLabels(getState());
-        if (!provider) return;
+const setMetadata =
+    ({
+        provider,
+        fileName,
+        data,
+    }: {
+        provider: MetadataProvider;
+        fileName: string;
+        data: WalletLabels | AccountLabels | undefined;
+    }) =>
+    (dispatch: Dispatch) => {
+        dispatch({
+            type: METADATA.SET_DATA,
+            payload: {
+                provider,
+                data: {
+                    [fileName]: JSON.parse(JSON.stringify(data)),
+                },
+            },
+        });
+    };
 
+export const getLabelableEntitities =
+    (deviceState: string) => (_dispatch: Dispatch, getState: GetState) => {
+        const { accounts } = getState().wallet;
+        const { devices } = getState();
+
+        return [
+            ...accounts
+                .filter(a => a.deviceState === deviceState)
+                .map(account => ({
+                    ...account.metadata,
+                    key: account.key,
+                    type: 'account' as const,
+                })),
+            ...devices
+                .filter((device: TrezorDevice) => device.state === deviceState)
+                .map((device: TrezorDevice) => ({
+                    ...device.metadata,
+                    state: device.state,
+                    type: 'device' as const,
+                })),
+        ];
+    };
+
+type LabelableEntity = ReturnType<ReturnType<typeof getLabelableEntitities>>[number];
+
+export const fetchMetadata =
+    ({
+        provider,
+        entity,
+        encryptionVersion = METADATA.ENCRYPTION_VERSION,
+    }: {
+        deviceStateArg?: string;
```

Comment:
not used here

Reply:
nice catch d08c54b652e7f89f441c7bfdc6a77dd481bda0a0